core/controllers/base.py

```diff
@@ -264,7 +264,9 @@ def dispatch(self):
         request_split = python_utils.url_split(self.request.uri)
         # If the request is to the old demo server, redirect it permanently to
         # the new demo server.
-        if request_split.netloc == 'oppiaserver.appspot.com':
+        # Note: this must use the https check (rather than e.g.
+        # request_split.netloc) since cron job requests in App Engine use http.
+        if self.request.uri.startswith('https://oppiaserver.appspot.com'):
```

Comment:
This solution doesn't feel right to me. If `oppiaserver.appspot.com` is just the old domain, then shouldn't we fix the cron jobs to use the new domain? The solution here is to not redirect HTTP requests, presumably because only the cron jobs access Oppia over HTTP. While this approach might work, the HTTP assumption seems unnecessarily complex.

I recognize that this creates an urgent problem by blocking the cron jobs, so if the "correct" solution will take too long to fix, can we open an issue to describe what that fix would be and track it?

Reply:
I don't think the cron jobs can be fixed to use a non-appspot.com domain, since their target URL is generated internally by App Engine. So, I think that we are stuck with this as the ultimate "correct" fix.

Though one alternative I can think of is to change the previous code instead to something like "if request_split.netloc == ... and url doesn't contain 'cron'". Do you think that would be better/sufficient?

Reply:
Yes, I think that would be better. Then at least the code would make clear that this is an intentional carve-out for cron jobs. Is it possible to have a narrower check than just whether the URL contains `cron`? For example, if a user chooses a username of `cron`, we don't want their profile page to get treated like a cron job

Reply:
Yup. I can look specifically for the prefix, so only for URLs that start with /cron.

Reply:
Done. PTAL.